R/DMMParam.R

```diff
@@ -0,0 +1,203 @@
+#' DMM clustering
+#' 
+#' Run the \code{DMM} function on a distance matrix 
+#' within \code{\link{clusterRows}}.
+#' 
+#' @param k An integer indicating the number of clusters, or a list of integers 
+#'   defining the possible number of clusters to choose from.
+#' @param type A string specifying the fit function to use to find the optimal 
+#'   number of clusters. Use in combination with k being a list of integers. 
+#'   Must be equal to \code{laplace}, \code{AIC} or \code{BIC}.
+#' @param transposed Logical scalar, is x transposed with samples in rows?
+#' @param seed An integer specifying the seed to use when doing the DMM 
+#'   algorithm. Will be random if not set.
+#' 
+#' @inheritParams clusterRows
+#' 
+#' @author Basil Courbayre
+#'
+#' @references
+#' Ian Holmes, Keith Harris, Christopher Quince (2012)
+#' Dirichlet multinomial mixtures: generative models for microbial metagenomics.
+#' \url{https://pubmed.ncbi.nlm.nih.gov/22319561/}
+#' 
+#' @details
+#' The DMM algorithm (see Holmes et al. 2012) is commonly used in microbial ecology along with metagenomic & 16S rRNA count data.
+#' 
+#' If we have microbiota data (for example the Twins data set in the DMM package), the suitable algorithm can be DMM.
+#' To modify an existing DMMParam object \code{x},
+#' users can simply call \code{x[[i]]} or \code{x[[i]] <- value} where \code{i} is any argument used in the constructor.
+#'
+#' If \code{k=NULL}, the cluster search range will be 1:3.
+#' The default for the fit function is \code{laplace}.
+#' 
+#' @return 
+#' The \code{DMMParam} constructor will return a \linkS4class{DMMParam} object with the specified parameters.
+#'
+#' The \code{clusterRows} method will return a factor of length equal to \code{nrow(x)} containing the cluster assignments.
+#' If \code{full=TRUE}, a list is returned with \code{clusters} (the factor, as above) and \code{objects} 
+#' (a list containing \code{dmm}, the dmm object; \code{k}, the number of clusters; 
+#' \code{prob}, the array of probabilities; and \code{seed} the seed for the dmm).
+#'
+#' @examples
+#' clusterRows(iris[,1:4], DMMParam())
+#' clusterRows(iris[,1:4], DMMParam(k=2))
+#' \dontrun{
+#' fl <- system.file(package="DirichletMultinomial", "extdata", "Twins.csv")
+#'         counts <- t(as.matrix(read.csv(fl, row.names=1)))
+#'         clusterRows(counts, DMMParam(k=1:3, type="laplace"))
+#' }
+#' # Example using a SummarizedExperiment
```

Comment:
What's the point of this example? It introduces a `Suggests` dependency on **mia** so it better be important, but it doesn't illustrate anything new with respect to how the function is used.

Reply:
The point is to have an example using a SE. We felt it was important to have one as this algorithm will often be used in this context.
Do you have another SE dataset that's already included in bluster perhaps? Or should I remove the example completely?

Reply:
I think it would be preferable if that example lived somewhere else. This is slightly too application-specific for **bluster**, which should keep its focus on the algorithms.

Indeed, the current `Suggests` dependency on the **scRNAseq** package is probably a mistake. The concrete applications for scRNA-seq should be discussed elsewhere, e.g., in the OSCA book.

You can mention the possible uses in the documentation, though.